src/proxyp/Header.cc

```diff
@@ -24,66 +23,69 @@ ProxyProtocol::Header::Header(const SBuf &ver, const Two::Command cmd):
 SBuf
 ProxyProtocol::Header::toMime() const
 {
-    SBufStream result;
+    SBuf result;
+    PackableStream os(result);
     for (const auto fieldType: EnumRange(Two::htPseudoBegin, Two::htPseudoEnd)) {
         const auto value = getValues(fieldType);
         if (!value.isEmpty())
-            result << PseudoFieldTypeToFieldName(fieldType) << ": " << value << "\r\n";
+            os << PseudoFieldTypeToFieldName(fieldType) << ": " << value << "\r\n";
     }
     // cannot reuse Header::getValues(): need the original TLVs layout
     for (const auto &tlv: tlvs)
-        result << tlv.type << ": " << tlv.value << "\r\n";
-    return result.buf();
+        os << tlv.type << ": " << tlv.value << "\r\n";
+    return result;
 }
 
 SBuf
 ProxyProtocol::Header::getValues(const uint32_t headerType, const char sep) const
 {
+    static char ipBuf[MAX_IPSTRLEN];
+    SBuf result;
+    PackableStream os(result);
+
     switch (headerType) {
 
     case Two::htPseudoVersion:
-        return version_;
+        os << version_;
+        break;
```

Comment:
Why was this method rewritten so much? It did not use SBufStream except in one special `switch` clause AFAICT. I see problems with some of the changes here, but I wonder if we can avoid evaluating them one by one by undoing virtually all of (i.e. minimizing) these changes instead?

Reply:
It is the end result of a long series of small code simplification refactors. There are several similar situations where things could improve further with the caller passing a stream in, but I have tried to scope this PR as only altering the SBuf API. Changing those others left to later discussions/PR.

Reply:
> It is the end result of a long series of small code simplification refactors

Please undo. These out-of-scope changes do not simplify the code in most cases and make it worse in some cases.

Reply:
Undone.

However FTR, all cases of this switch statement except 'htPseudoVersion' perform SBuf allocation to produce the returned object. All the ToSBuf() calls on return statements hide inefficient uses of PackableStream to serialize their parameters into another SBuf.

Reply:
Thank you for undoing these changes.

> all cases of this switch statement except 'htPseudoVersion' perform SBuf allocation

Translation: The PR was adding one more allocation (without disclosure/justification).

> All the ToSBuf() calls on return statements hide inefficient uses of PackableStream to serialize their parameters into another SBuf.

Could you clarify this claim, please? ToSBuf() explicitly tells the code reader that a conversion to SBuf is going on. Where do you see the official method code _hiding_ its inefficiencies?